osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -95,6 +95,22 @@ public abstract class ScoreProcessor
         /// </summary>
         protected virtual bool DefaultFailCondition => Health.Value == Health.MinValue;
 
+        private bool adjustRank;
+
+        /// <summary>
+        /// Used by specific mods to adjust <see cref="Rank"/>.
+        /// </summary>
+        public bool AdjustRank
```

Comment:
You are going to have to explain why this needs to exist so we can advise you on a better way of doing it.

Reply:
Alright, so this boolean is used by specific mods (Hidden mod for now) to tell the score processor to do adjustments on the grade calculated by Accuracy, there was another idea of that suggested by smoogipoo is to create a function that changes `ScoreProcessor.Rank` value by the mod itself but the problem was where to invoke it

Reply:
The reason why I suggested the function being a part of a mod is because I see mods potentially doing more than just applying SH and XH. For example, they could define a completely different scoring algorithm, with different grade points to reflect that.

The invocation would be done from inside `ScoreProcessor`.

1. `ScoreProcessor` would receive mods from `Player`, via constructor.
2. `ScoreProcessor` would invoke:  

    ```
    Mods.OfType<IApplicableToScoreProcessor>().ForEach(m => rank = m.AdjustRank(rank));
    ```